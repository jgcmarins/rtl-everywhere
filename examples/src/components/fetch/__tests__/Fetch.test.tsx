import React from 'react'
import { render, waitForElement } from '@testing-library/react';

import Fetch from '../Fetch';

const ATOM_URL = 'https://github.githubassets.com/images/icons/emoji/atom.png?v8';

window.fetch = jest.fn().mockImplementation(() => {
  return new Promise((resolve, reject) => {
    resolve({
      json: jest.fn().mockImplementation(() => {
        return new Promise((resolve, reject) => {
          resolve({ atom: ATOM_URL });
        });
      })
    });
  });
});

describe('Fetch Tests', () => {
  const { getByTestId, container } = render(<Fetch/>);

  it('should render Fetch Component', async () => {
    const spanTextNode = await waitForElement(() =>
      getByTestId('fetch-head'),
    );

    expect(spanTextNode.textContent).toBe('Fetch');
  });

  it('should check count value on first render', async () => {
    // https://github.com/testing-library/react-testing-library/issues/281

    expect(container.firstChild).toMatchSnapshot()
  });
});