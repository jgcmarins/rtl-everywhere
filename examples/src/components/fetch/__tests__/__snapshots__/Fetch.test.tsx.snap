// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetch Tests should check count value on first render 1`] = `
<div
  class="sc-bdVaJa kLZEci"
>
  <span
    data-testid="fetch-head"
  >
    Fetch
  </span>
  <img
    data-testid="fetch-img"
    src="https://github.githubassets.com/images/icons/emoji/atom.png?v8"
  />
</div>
`;
